workflow/validate/validate.go

```diff
@@ -1017,6 +1049,28 @@ func validateOutputParameter(paramRef string, param *wfv1.Parameter) error {
 	return nil
 }
 
+// getNameFieldValue gets the value of the `Name` field from the struct
+// if not found from the struct, it checks embedded field instead
+func getNameFieldValue(s reflect.Value) (string, error) {
+	for i := 0; i < s.NumField(); i++ {
+		typeField := s.Type().Field(i)
+		if typeField.Name == "Name" {
+			return s.Field(i).String(), nil
+		}
+	}
+	// if not found, look for embedded struct fields
+	for i := 0; i < s.NumField(); i++ {
+		typeField := s.Type().Field(i)
+		if typeField.Anonymous {
+			name, err := getNameFieldValue(s.Field(i))
+			if err == nil {
+				return name, nil
+			}
+		}
+	}
+	return "", errors.InternalError("No 'Name' field in struct")
+}
```

Comment:
The original `getNameFieldValue` doesn't support `Name` field being embedded in another struct. So I updated it in order to use it for `ContainerNode` struct. For example:

```go
type Container struct {
	Name string
}

type ContainerNode struct {
	Container
}
```

Moved the func out so that I can write unit tests for it.